rust/template/d3log/src/json_framer.rs

```diff
@@ -0,0 +1,96 @@
+// this module pre-frames json objects out of an arbitrarily chunked byte string.
+// it is a really terrible idea to temporarily work around some limtations
+// with serde. specifically that it can read chunked blocks and a sequence
+// of self-framed json objects, but not both
+//
+// haven't been able to understand it yet but
+// https://github.com/TimelyDataflow/timely-dataflow/blob/master/timely/src/dataflow/operators/to_stream.rs#L73
+// may lead to a better answer.
+//
+// if it were going to stay Lexement should really just be the leading group
+// character
+
+use crate::error::Error;
+use phf::{phf_map, phf_set};
+
+#[allow(dead_code)]
+static STARTS: phf::Set<char> = phf_set! {
+    '[',
+    '{',
+    '"',
+};
```

Comment:
Using `phf` is a bit of a "sledgehammer for a fruitfly" situation, we can just use a byte array (`[u8; 3]` or `&[u8]`, either works) and then use `.contains()` on it, if LLVM doesn't play ball then we can just manually make a jumptable *if required*, but I'd rather see a working prototype before micro-optimizing it